### ISSUE TITLE
Print request headers regardless of connection error

### ISSUE THREAD

**Issue by maciej** [jakubroztocil/httpie#639]
If `-v` is set in the command line arguments HTTPie prints the request along with the response.

The request is however not printed if a connection error happened, eg. the connection was closed by server before receiving a response.

I think it would be helpful to see the request printed in such case for debugging purposes.

```
> http -v GET http://127.0.0.1:1234/
http: error: ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)) while doing GET request to URL: http://127.0.0.1:1234/
```

In the above case the server closed the connection _after_ the request was completely sent.

Running httpie 0.9.9, detailed debug output attached:
[httpie-debug.txt](https://github.com/jakubroztocil/httpie/files/1539173/httpie-debug.txt) attached

**Comment by dejicranium** [comment]
Can I submit a pull request for this feature?

**Comment by maciej** [comment]
@dejicranium as the issue reporter I have nothing against you submitting a PR. :-)

**Comment by jakubroztocil** [comment]
This is definitely a wanted feature. It applies to the request body as well. It should also be possible to just construct an HTTP request and print it out without actually making it, something `http --dry-run example.org`. 

I'm afraid, though, that this will need a larger refactoring of the output machinery. Functionality affected by `--all`, `--download`,  etc. will need to be updated. 

I'm working (very slowly) on streaming uploads (when done, v1.0.0 will go out), and for that I'm changing that part of the codebase quite a bit. I'll see if I can squeeze in this feature as well.

**Comment by jakubroztocil** [comment]
This has now been pushed to `master` in bece3c77bb51ecc55dcc4008375dc29ccd91575c) and it will be released when v2.0.0 comes out. Each request and response is now handled independently as they are available. Also added `--offline` for dry runs.
